app/frontend/controllers/tabs_controller.js

```diff
@@ -0,0 +1,30 @@
+document.addEventListener("DOMContentLoaded", function() {
+  const urlParams = new URLSearchParams(window.location.search);
+  const tab = urlParams.get("tab");
+
+  function activateTab(tabId) {
+    const tabElement = document.getElementById(tabId);
+    if (tabElement) {
+      tabElement.click();
+    }
+    history.replaceState(null, null, `${window.location.pathname}?${urlParams.toString()}`);
```

Comment:
This line:

- Removes the anchor: The fragment identifier (anchor) is not included in the new URL, effectively removing it.
- Keeps the tab URL parameter: The urlParams object, which contains the tab parameter, is converted to a query string and appended to the path, ensuring the tab parameter remains in the URL.